### PR TITLE
set up random seed for reproducible results

### PR DESCRIPTION
**Pull request by Farnazmdi**
In this PR:
- Added random init seed for state distributions (E) and when we initialize the model parameters (T,pi).
The tests still occasionally fails..

### CODE REVIEW

**Comment by aarmey** [comment]
The test needs to initialize and then pass in the rng to ensure it is set to the same state. I am not seeing anything set within the tests that are failing...

File: lineage/Analyze.py [hunk 1]
```diff
@@ -103,13 +103,14 @@ def Analyze_list(
...
 
     for _ in range(5):
         tHMMobj_list2 = [
             tHMM(X, num_states=num_states, **kwargs) for X in pop_list
         ]  # build the tHMM class with X
-        _, _, _, gammas2, LL2 = fit_list(tHMMobj_list2)
+        _, _, _, gammas2, LL2 = fit_list(tHMMobj_list2, rng)
 
         if LL2 > LL:
             tHMMobj_list = tHMMobj_list2
```
Comment (Farnazmdi) [inline lineage/Analyze.py:116]: @aarmey In this function, we run the fitting multiple times and choose the best fit. If we have the random seed here, this multiple running would be meaningless, and also, if we don't set a random seed and run this fitting multiple times, then there is no guarantee it finds the same answers.. 
Should we have only 1-time fitting here?

Comment (aarmey) [reply lineage/Analyze.py:116]: If you are passing the `rng` around, the seed will be different each fit, but the collection of them will be reproducible. e.g., the first time it'll be 1, second time 6, third time 23, etc.

File: lineage/tHMM.py [hunk 2]
```diff
@@ -19,19 +20,20 @@ def __init__(self, X: list[LineageTree], nState: int, fpi=None, fT=None, fE=None
...
         """
+
         self.fpi = fpi
         self.fT = fT
         self.fE = fE
         self.num_states = nState
 
         if self.fpi is None or self.fpi is True:
-            self.pi = np.random.rand(nState)
+            self.pi = sp.uniform.rvs(0.0, 1.0, nState)
```
Comment (aarmey) [inline lineage/tHMM.py:30]: This isn't using the random number generator. You can use [`rng.uniform()`](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.uniform.html) here.

File: lineage/tHMM.py [hunk 3]
```diff
@@ -19,19 +20,20 @@ def __init__(self, X: list[LineageTree], nState: int, fpi=None, fT=None, fE=None
...
             self.pi /= np.sum(self.pi)
         else:
             self.pi = self.fpi
 
         if self.fT is None:
-            self.T = np.random.dirichlet(np.random.rand(nState), nState)
+            self.T = sp.dirichlet.rvs(np.ones(nState), size=nState, random_state=rng)
```
Comment (aarmey) [inline lineage/tHMM.py:36]: [`rng.dirichlet()`](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.dirichlet.html) doesn't need scipy.

**Comment by aarmey** [comment]
@Farnazmdi this should now be working. Please look this over then I can merge.

**Comment by Farnazmdi** [comment]
Thank you so much..
Just one question. Does it matter that we use the same random seed number in all files? Or it just matters that those numbers, in any function, stay the same when we run the functions at different times?

**Comment by aarmey** [comment]
I would suggest you read through the numpy documentation, and think about how this works within the context of the analysis. This is something you should be able to answer at this point.

**Pull request merged**